src/mainmenu.asm

```diff
@@ -905,7 +1108,7 @@ rng_rerandomize:
 
 rng_botwoon_rng:
     dw !ACTION_CHOICE
-    dl #$7FFB8A
+    dl #!ram_botwoon_rng
```

Comment:
$7FFB8A was the address used in the original implementation for v1.42. It was moved to $7FFB86 in defines.asm. This was only half the problem.

The old menu (v1.42) wrote specific values for each choice, while the current menu can only pass the menu index. hook_botwoon_rng needs to convert the index to those specific values, or branch out if zero (set to random). More on that in the rng.asm comment.